src/Core/Resolvers/MsSqlQueryExecutor.cs

```diff
@@ -83,6 +85,49 @@ public MsSqlQueryExecutor(
             ConfigureMsSqlQueryEecutor();
         }
 
+        /// <summary>
+        /// Creates a SQLConnection to the data source of given name. This method also adds an event handler to
+        /// the connection's InfoMessage to extract the statement ID from the request and add it to httpcontext.
+        /// </summary>
+        /// <param name="dataSourceName">The name of the data source.</param>
+        /// <returns>The SQLConnection</returns>
+        /// <exception cref="DataApiBuilderException">Exception thrown if datasource is not found.</exception>
+        public override SqlConnection CreateConnection(string dataSourceName)
+        {
+            if (!ConnectionStringBuilders.ContainsKey(dataSourceName))
+            {
+                throw new DataApiBuilderException("Query execution failed. Could not find datasource to execute query against", HttpStatusCode.BadRequest, DataApiBuilderException.SubStatusCodes.DataSourceNotFound);
+            }
+
+            SqlConnection conn = new()
+            {
+                ConnectionString = ConnectionStringBuilders[dataSourceName].ConnectionString,
+            };
+
+            // If connection is a SQLConnection and it is not null, we can extract the info message
+            conn.InfoMessage += (object sender, SqlInfoMessageEventArgs e) =>
+            {
+                // Log the statement ids returned by the SQL engine when we executed the batch.
+                // This helps in correlating with SQL engine telemetry.
+
+                // If the info message has an error code that matches the well-known codes used for returning statement ID,
+                // then we can be certain that the message contains no PII.
+                IEnumerable<SqlError> errorsReceived = e.Errors.Cast<SqlError>();
```

Comment:
The same executor is used for SQL Server on premises, can we please look make sure at least all flavors of mssql have this change. 

As a followup change, I'd expect PgSql, Mysql QueryExecutors implement this function as well to complete the feature - otherwise its a broken functionality

Reply:
Discussed on teams. 
Created an action item to follow up on the implementations for PgSql and Mysql QueryExecutors in a separate PR.

Regarding the first part of this comment, please let me know if I missed any other mssql query executor file.